pkg/storage/postgres/postgres.go

```
@@ -0,0 +1,322 @@
+		VALUES ($1, $2, $3, $4, $5, $6)
+	`, record.GetType(), record.GetId(), record.GetVersion(), data, modifiedAt, deletedAt)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func putRecord(ctx context.Context, q querier, record *databroker.Record) error {
+	data := jsonbFromAny(record.GetData())
+	modifiedAt := timestamptzFromTimestamppb(record.GetModifiedAt())
+	var err error
+	if record.GetDeletedAt() == nil {
+		_, err = q.Exec(ctx, `
+			INSERT INTO `+schemaName+`.`+recordsTableName+` (type, id, version, data, modified_at)
+			VALUES ($1, $2, $3, $4, $5)
+			ON CONFLICT (type, id) DO UPDATE
+			SET version=$3, data=$4, modified_at=$5
+		`, record.GetType(), record.GetId(), record.GetVersion(), data, modifiedAt)
+	} else {
+		_, err = q.Exec(ctx, `
+			DELETE FROM `+schemaName+`.`+recordsTableName+`
+			WHERE type=$1 AND id=$2 AND version<$3
+		`, record.GetType(), record.GetId(), record.GetVersion())
+	}
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func setOptions(ctx context.Context, q querier, recordType string, options *databroker.Options) error {
+	capacity := pgtype.Int8{Status: pgtype.Null}
+	if options != nil && options.Capacity != nil {
+		capacity.Int = int64(options.GetCapacity())
+		capacity.Status = pgtype.Present
+	}
+
+	_, err := q.Exec(ctx, `
+		INSERT INTO `+schemaName+`.`+recordOptionsTableName+` (type, capacity)
+		VALUES ($1, $2)
+		ON CONFLICT (type) DO UPDATE
+		SET capacity=$2
+	`, recordType, capacity)
+	return err
+}
+
+func signalRecordChange(ctx context.Context, q querier) error {
+	_, err := q.Exec(ctx, `NOTIFY `+recordChangeNotifyName)
+	return err
+}
+
+func jsonbFromAny(any *anypb.Any) pgtype.JSONB {
+	if any == nil {
+		return pgtype.JSONB{Status: pgtype.Null}
+	}
+
+	bs, err := protojson.Marshal(any)
+	if err != nil {
+		log.Warn(context.Background()).
+			Err(err).
+			Str("data", prototext.Format(any)).
+			Msg("storage/postgres: error marshaling any to JSON")
+		return pgtype.JSONB{Status: pgtype.Null}
```

shouldn't we bubble that error up and fail record update?